core/utils.py

```diff
@@ -91,6 +92,28 @@ class ExplorationConversionError(Exception):
     pass
 
 
+def open_file(
+        filename: str, mode: str, encoding: Union[str, None]='utf-8',
+        newline: Union[str, None]=None) -> io.TextIOWrapper:
+    """Open file and return a corresponding file object.
+
+    Args:
+        filename: str. The file to be opened.
+        mode: str. Mode in which the file is opened.
+        encoding: str. Encoding in which the file is opened.
+        newline: None|str. Controls how universal newlines work.
+
+    Returns:
+        _io.TextIOWrapper. The file object.
+
+    Raises:
+        FileNotFoundError. The file cannot be found.
+    """
+    file = open(filename, mode, encoding=encoding, newline=newline)
```

Comment:
This seems different from the previous implementation in https://github.com/oppia/oppia/pull/14728/files?file-filters%5B%5D=.py&owned-by%5B%5D=seanlip&show-viewed-files=true#diff-c75d2b5dd27e992aa9d77c5849affa12cc8a81e304fc131fde9bad55763bd9bfL150 . Why is that?

Reply:
io.open is just an alias for in-build open function . Also since migrating to python3 every function is required to have annotations , the return type annotation for open_file was giving error during mypy checks . So i just used cast to change the return type to my specific type ( textIOWrapper ) .

Reply:
In general, we prefer to avoid casting. Can you explain, conceptually, why the general type-check was failing (not just "it failed mypy checks")? For example, how do we know that the return type here shouldn't be something else? At the very least there should be a detailed comment explaining the reasons for the cast.

/cc @vojtechjelinek -- do you have any perspective on this?

Reply:
Hey, I think that here we should use `TextIO` from `typing` here, can you try it?

Reply:
Sure i will address to both your remarks @vojtechjelinek @seanlip and will try to solve this mypy check failing .